### PR TITLE
feat: add Azure Managed Identities attributes for ARO-HCP Clusters

### PR DESCRIPTION
**Pull request by miguelsorianod**
We add Azure Managed Identities related attributes that are going to be leveraged for the ARO-HCP Clusters.

### CODE REVIEW

**Comment by miguelsorianod** [comment]
@machi1990 I think it should be fine to add this even when we don't have any code in CS to map in/out.

Would you agree?

**Comment by machi1990** [comment]
> @machi1990 I think it should be fine to add this even when we don't have any code in CS to map in/out.
> 
> My understanding is that if the new content of the OCM SDK Go is leveraged:
> 
> * When creating the type using the OCM SDK Go and it is sent to CS then CS will deserialize into the CS specific API type and nothing else will happen as there are no validations nor mapping to model layer.
> * When reading from CS and serializing into the OCM SDK Go type the attributes won't exist from CS so the corresponding attributes in the OCM Go SDK type will not be set or set to zero-values
> 
> Would you agree?

Yes. That's what will happen. In my opinion, it should be safe to roll this one out once review. 

I've added it to my list and I'll get to reviewing it at some point today.

**Review by machi1990 — APPROVED** [review]
lgtm

Spotted one typo and left a corrective suggestion. Good to merge once addressed.

File: model/clusters_mgmt/v1/azure_operators_authentication_types.model [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	ServiceManagedIdentity AzureServiceManagedIdentity
+}
+
+// Represents the information associated to an Azure User-Assigned
+// Managed Identity belonging to the Control Plane of the cluster.
+struct AzureControlPlaneManagedIdentity {
+	// The Azure Resource ID of the Azure User-Assigned Managed
+	// Identity. The managed identity represented must exist before
+	// creating the cluster.
+	// he Azure Resource Group Name specified as part of the Resource ID
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply model/clusters_mgmt/v1/azure_operators_authentication_types.model]: :+1: updated

**Review by machi1990 — APPROVED** [review]

**Pull request merged**